doc/api.yaml

```diff
@@ -0,0 +1,113 @@
+openapi: 3.0.2
+
+info:
+  title: MockIO
+  version: 1.0.0
+  contact:
+    name: Source Busters
+  description: |
+    Configurable mock server for mocking HTTP requests.
+
+
+paths:
+  /config:
+
+    post:
+      summary: Create a named configurable mock instance
+
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/MockParams"
+
+      responses:
+        "204":
+          description: Mock instance was created successfully
+
+        "404":
+          description: Mock parameters validation failed
+
+
+
+  /config/{mockId}:
```

Comment:
add a get to pull down current mock config? be useful for debugging

Reply:
we can add it later, just want to have the bare minimum for the service to be usable